WorkflowSwiftUI/Sources/ObservableScreen.swift

```diff
@@ -0,0 +1,208 @@
+#if canImport(UIKit)
+
+import SwiftUI
+import Workflow
+import WorkflowUI
+
+/// A screen that renders SwiftUI views with an observable model for fine-grained invalidations.
+///
+/// Screens conforming to this protocol will render SwiftUI views that observe fine-grained changes
+/// to the underlying model, and selectively invalidate in response to changes to properties that
+/// are accessed by the view.
+///
+/// Invalidations happen when the observed state is mutated, during actions or the
+/// `workflowDidChange` method. When this screen is rendered, a new model is injected into the
+/// store. Any invalidated views will then be updated with the new model by SwiftUI during its own
+/// rendering cycle.
+///
+/// To use this protocol with a workflow, your workflow should render a type that conforms to
+/// ``ObservableModel``, and then mapped to a screen implementation that uses that concrete model
+/// type. See ``ObservableModel`` for options on how to render one easily.
+public protocol ObservableScreen: Screen {
+    /// The type of the root view rendered by this screen.
+    associatedtype Content: View
+    /// The type of the model that this screen observes.
+    associatedtype Model: ObservableModel
+
+    /// The sizing options for the screen.
+    var sizingOptions: SwiftUIScreenSizingOptions { get }
+    /// The model that this screen observes.
+    var model: Model { get }
+
+    /// Constructs the root view for this screen. This is only called once to initialize the view.
+    /// After the initial construction, the view will be updated by injecting new values into the
+    /// store.
+    @ViewBuilder
+    static func makeView(store: Store<Model>) -> Content
```

Comment:
Can/how/should this have access to the Environment?

Reply:
(In general, this being static feels a bit weird to me, I assume it's to reduce access to instance properties, but is that a hard requirement?)

Reply:
Got it, makes sense – curious to see how this is received by folks, since feels a bit weird, but I have no evidence for that!